### PR TITLE
fix(components): JOB-61635 InputNumber on iOS not allowing decimals

### PR DESCRIPTION
**Pull request by MichaelParadis**
### Added

- 

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- Fixed in ability to add decimals on iOS for mobile because the inputMode wasn't set https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/inputmode

### Security

- 

## Testing

- Open on iOS, set `keyboard` to `decimal`. Verify that it is possible to see the decimal on the keyboard

---
IMG_93A4C4444CF0-1

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/InputNumber/InputNumber.tsx [hunk 1]
```diff
@@ -18,6 +18,11 @@ interface InputNumberProps
       | "defaultValue"
     > {
   value?: number;
+
+  /**
+   * Allow keyboards for mobile to display decimal option
+   */
+  allowDecimals?: boolean;
```
Comment (chris-at-jobber) [inline packages/components/src/InputNumber/InputNumber.tsx:25]: Would allowing this component to `Pick` the keyboard prop with a default of `numeric` have the same effect?

Comment (MichaelParadis) [reply packages/components/src/InputNumber/InputNumber.tsx:25]: Actually, I think that may would be better. Let me update things

**Comment by chris-at-jobber** [comment]
I'm able to test that the correct keyboards are called when the value is changed.

One note (possibly for future investigation as this can be replicated on production): When I have typed a `.` in the InputNumber, the minilabel collapses down into overlapping with the entered value: 

https://user-images.githubusercontent.com/39704901/216725243-db1fc116-f64e-4d87-8db1-2991dadceed1.mov

**Comment by dhaberst** [comment]
@MichaelParadis @chris-at-jobber This lgtm!

**Pull request merged**